components/src/dynamo/vllm/handlers.py

```
@@ -625,6 +625,280 @@ async def wake_up(self, body: dict) -> dict:
+                # already registered, so we must remove the previous adapter
+                # first. remove_lora is best-effort on a fresh add.
+                if is_hot_swap:
+                    old_id = self.loaded_loras[lora_name].id
+                    try:
+                        await self.engine_client.remove_lora(old_id)
+                        # Invalidate the cache entry immediately after remove succeeds.
+                        # If add_lora below fails, this prevents a stale entry pointing
+                        # at an adapter the engine no longer holds from poisoning future
+                        # rollouts with wrong importance ratios (Tier-1 RL correctness risk).
+                        self.loaded_loras.pop(lora_name, None)
+                    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)